Tools/autotest/common.py

```diff
@@ -493,10 +494,17 @@ def set_parameter(self, name, value, add_to_context=True):
                           % (returned_value, value))
         raise ValueError()
 
-    def get_parameter(self, name):
-        self.mavproxy.send("param fetch %s\n" % name)
-        self.mavproxy.expect("%s = ([-0-9.]*)\r\n" % (name,))
-        return float(self.mavproxy.match.group(1))
+    def get_parameter(self, name, num_rety=1, timeout=60):
+        for i in range(0, num_rety):
+            self.mavproxy.send("param fetch %s\n" % name)
+            try:
+                self.mavproxy.expect("%s = ([-0-9.]*)\r\n" % (name,), timeout=timeout/num_rety)
+                return float(self.mavproxy.match.group(1))
+            except pexpect.TIMEOUT:
+                if i < num_rety:
+                    continue
+                else:
```

Comment:
pointless else

Reply:
hum so we don't stop the test when we cannot get the parameter ?